test/unit/ngsiv2/expressions/expressionBasedTransformations-test.js

```diff
@@ -0,0 +1,833 @@
+                done();
+            });
+        });
+    });
+
+    describe('When an update comes for attributes without expressions and Object type', function() {
+        // Case: Update for a JSON array attribute without expression
+
+        var values = [
+            {
+                name: 'r',
+                type: 'Object',
+                value: ['v0.1', 'v0.2', 'v0.3']
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/light1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextExpressionPlugin7.json'))
+                .reply(204);
+        });
+
+        it('should apply the expression before sending the values', function(done) {
+            iotAgentLib.update('light1', 'Light', '', values, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When there are expressions including other attributes and they are not updated', function() {
+
+        var values = [
+            {
+                name: 'x',
+                type: 'Float',
+                value: 0.44
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/light1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextExpressionPlugin12.json'))
+                .reply(204);
+        });
+
+        it('should apply the expression before sending the values', function(done) {
+            iotAgentLib.update('light1', 'Light', '', values, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When there are expressions including other attributes and they are updated', function() {
+
+        var values = [
+            {
+                name: 'p',
+                type: 'Integer',
+                value: 10
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/light1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextExpressionPlugin13.json'))
```

Comment:
It seems that updateContextExpressionPlugin13.json is not part of the PR... Maybe you missed `git add` of that file before your last commit.

Reply:
Fixed in f50c6f43594efa5bd5122fd2c4c15bbbbe7c405d